pkg/controllers/machine/disruption/emptiness_test.go

```diff
@@ -0,0 +1,129 @@
+			},
+		})
+	})
+
+	It("should mark machines as empty", func() {
+		provisioner.Spec.TTLSecondsAfterEmpty = ptr.Int64(30)
+		node.Annotations = lo.Assign(node.Annotations, map[string]string{
+			v1alpha5.EmptinessTimestampAnnotationKey: fakeClock.Now().Format(time.RFC3339),
+		})
+		ExpectApplied(ctx, env.Client, provisioner, machine, node)
+
+		// step forward to make the node expired
+		fakeClock.Step(60 * time.Second)
+		ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKeyFromObject(machine))
+
+		machine = ExpectExists(ctx, env.Client, machine)
+		Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineEmpty).IsTrue()).To(BeTrue())
+	})
+	It("should remove the status condition from the machine when emptiness is disabled", func() {
+		machine.StatusConditions().MarkTrue(v1alpha5.MachineEmpty)
+		ExpectApplied(ctx, env.Client, provisioner, machine, node)
+
+		ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKeyFromObject(machine))
+
+		machine = ExpectExists(ctx, env.Client, machine)
+		Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineEmpty)).To(BeNil())
+	})
+	It("should remove the status conditions from the machine when the node doesn't exist", func() {
+		provisioner.Spec.TTLSecondsAfterEmpty = ptr.Int64(30)
+		machine.StatusConditions().MarkTrue(v1alpha5.MachineEmpty)
+		ExpectApplied(ctx, env.Client, provisioner, machine)
+
+		ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKeyFromObject(machine))
+
+		machine = ExpectExists(ctx, env.Client, machine)
+		Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineEmpty)).To(BeNil())
+	})
+	It("should remove the status conditions from the machine when the node doesn't have the emptiness timestamp", func() {
+		provisioner.Spec.TTLSecondsAfterEmpty = ptr.Int64(30)
+		machine.StatusConditions().MarkTrue(v1alpha5.MachineEmpty)
+		ExpectApplied(ctx, env.Client, provisioner, machine, node)
+
+		ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKeyFromObject(machine))
+
+		machine = ExpectExists(ctx, env.Client, machine)
+		Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineEmpty)).To(BeNil())
+	})
+	It("should remove the status conditions from the machine when the node timestamp can't be parsed", func() {
+		provisioner.Spec.TTLSecondsAfterEmpty = ptr.Int64(30)
+		node.Annotations = lo.Assign(node.Annotations, map[string]string{
+			v1alpha5.EmptinessTimestampAnnotationKey: "bad-value",
+		})
+		machine.StatusConditions().MarkTrue(v1alpha5.MachineEmpty)
+		ExpectApplied(ctx, env.Client, provisioner, machine, node)
+
+		ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKeyFromObject(machine))
+
+		machine = ExpectExists(ctx, env.Client, machine)
+		Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineEmpty)).To(BeNil())
+	})
+	It("should remove the status condition from non-empty machines", func() {
+		provisioner.Spec.TTLSecondsUntilExpired = ptr.Int64(200)
```

Comment:
This should be empty, right?
```suggestion
		provisioner.Spec.TTLSecondsAfterEmpty = ptr.Int64(200)
```